lib/geonames.ex

```diff
@@ -23,13 +23,21 @@ defmodule Geonames do
         token_param
       }"
 
-    case HTTPoison.get(url) do
+    # force TLS v1.2, see https://campezzi.ghost.io/httpoison-ssl-connection-closed/
+    case HTTPoison.get(url, [], ssl: [versions: [:"tlsv1.2"]]) do
       {:ok, %HTTPoison.Response{status_code: 200, body: body}} ->
         body
         |> Jason.decode!(strings: :copy)
 
       response ->
-        Logger.warn(fn -> "Unexpected response from #{url} : #{inspect(response)}" end)
+        sanitized_url =
+          if geonames_token do
+            String.replace(url, geonames_token, "SECRET")
```

Comment:
not strictly within the scope of this PR but how secret is our Geonames token anyhow? is it having been logged sufficient grounds to roll it?

Reply:
I think it's probably fine.